gridcomps/History3G/OutputInfo.F90

```diff
@@ -0,0 +1,349 @@
+         name = get_vertical_dim_spec_info(info(i), _RC)
+         if(find_index(names, name) == 0) call names%push_back(name)
+      end do
+      _RETURN(_SUCCESS)
+
+   end function get_vertical_dim_spec_names_bundle_info
+
+   function get_vertical_dim_spec_name_field(field, rc) result(spec_name)
+      character(len=:), allocatable :: spec_name
+      type(ESMF_Field), intent(inout) :: field
+      integer, optional, intent(out) :: rc
+      integer :: status
+      type(ESMF_Info) :: info
+
+      call ESMF_InfoGetFromHost(field, info, _RC)
+      spec_name = get_vertical_dim_spec_info(info, _RC)
+      call ESMF_InfoDestroy(info, _RC)
+      _RETURN(_SUCCESS)
+
+   end function get_vertical_dim_spec_name_field
+
+   function get_vertical_dim_spec_info(info, rc) result(spec_name)
+      character(len=:), allocatable :: spec_name
+      type(ESMF_Info), intent(in) :: info
+      integer, optional, intent(out) :: rc
+      integer :: status
+      integer :: n
+
+      spec_name = ''
+      n = get_num_levels_info(info, _RC) 
+      _RETURN_UNLESS(n > 0)
+      call ESMF_InfoGetCharAlloc(info, key=KEY_VLOC, value=spec_name, _RC)
+      _RETURN(_SUCCESS)
+
+   end function get_vertical_dim_spec_info
+
+   function get_ungridded_dims_bundle(bundle, tol, rc) result(dims)
+      type(UngriddedDims) :: dims
+      type(ESMF_FieldBundle), intent(in) :: bundle
+      real, optional, intent(in) :: tol
+      integer, optional, intent(out) :: rc
+      integer :: status
+      type(ESMF_Info), allocatable :: info(:)
+      type(UngriddedDimVector) :: vec
+      real :: tol_
+
+      tol_ = 1E-8
+      if(present(tol)) tol_ = tol
+      info = create_bundle_info(bundle, _RC)
+      vec = get_ungridded_dims_bundle_info(info, tol_, _RC)
+      dims = UngriddedDims(vec)
+      call destroy_bundle_info(info, _RC)
+      _RETURN(_SUCCESS)
+
+   end function get_ungridded_dims_bundle
+
+   function get_ungridded_dims_bundle_info(info, tol, rc) result(vec)
+      type(UngriddedDimVector) :: vec
+      type(ESMF_Info), intent(in) :: info(:)
+      real, intent(in) :: tol
+      integer, optional, intent(out) :: rc
+      integer :: status
+      integer :: i
+
+      vec = UngriddedDimVector()
+      do i=1, size(info)
+         call push_ungridded_dim(vec, info(i), tol, _RC)
+      end do
+      _RETURN(_SUCCESS)
+
+   end function get_ungridded_dims_bundle_info
+
+   function get_ungridded_dims_field(field, tol, rc) result(ungridded)
+      type(UngriddedDims) :: ungridded
+      type(ESMF_Field), intent(inout) :: field
+      real, optional, intent(in) :: tol
+      integer, optional, intent(out) :: rc
+      integer :: status
+      type(ESMF_Info) :: info
+      type(UngriddedDimVector) :: vec
+      real :: tol_
+
+      tol_ = 1E-8
+      if(present(tol)) tol_ = tol
+
+      call ESMF_InfoGetFromHost(field, info, _RC)
+      call push_ungridded_dim(vec, info, tol_, _RC)
+      ungridded = UngriddedDims(vec)
+      call ESMF_InfoDestroy(info, _RC)
```

Comment:
I think GetFromHost returns a shallow copy.  Destroy is probably bad here.

Reply:
True. Object handle. Destroy removed.